bogdann98_task3/DynamicStringArray.cpp

```diff
@@ -0,0 +1,124 @@
+#include "DynamicStringArray.h"
+// default constructor that sets the dynamic array to nullptr and sets size to 0
+DynamicStringArray::DynamicStringArray() : size{0}, dynamicArray{nullptr}
+{
+}
+
+// a function that return size
+int DynamicStringArray::getSize() const
+{
+    return size;
+}
+
+// function creates a new dynamic array one element larger than dynamicArray and copy all element into the new array
+void DynamicStringArray::addEntry(const string &String)
+{
+    string *newdynamicArray = new string[size + 1];
+    for (int i = 0; i < size; i++)
+    {
+        newdynamicArray[i] = dynamicArray[i];
+    }
+    newdynamicArray[size] = String;
+    delete[] dynamicArray;
+    dynamicArray = newdynamicArray;
+    size++;
+}
+// the function boolean should search dynamicArray for the string
+bool DynamicStringArray::deleteEntry(const string &String)
+{
+    bool found = 0;
+    if (dynamicArray[0] == String && size == 1)
+    {
+        delete[] dynamicArray;
+        size--;
+        found = 1;
+    }
+    if (dynamicArray[0] == String && size > 1)
```

Comment:
So if the String is not found you will try anyway to modify the dynamicArray?
Can you please add a test where you delete a string that is not in the dynamicArray and see what happen.